server/src/main/kotlin/delta/codecharacter/server/daily_challenge/DailyChallengeEntity.kt

```diff
@@ -1,13 +1,17 @@
 package delta.codecharacter.server.daily_challenge
 
+import com.fasterxml.jackson.annotation.JsonProperty
 import delta.codecharacter.dtos.ChallengeTypeDto
+import org.springframework.data.annotation.Id
 import org.springframework.data.mongodb.core.mapping.Document
+import java.util.UUID
 
 @Document(collection = "daily_challenges")
 data class DailyChallengeEntity(
-    val day: Int,
-    val challName: String,
-    val challType: ChallengeTypeDto,
-    val chall: String,
-    val description: String?,
+        @Id val id: UUID,
+        @field:JsonProperty("day") val day: Int,
+        @field:JsonProperty("challName") val challName: String,
+        @field:JsonProperty("challType") val challType: ChallengeTypeDto,
+        @field:JsonProperty("chall") val chall: String,
+        @field:JsonProperty("description") val description: String?,
```

Comment:
No need of adding @field property if we are using a `separate` data class for deserialization.

@field:JsonProperty is to match with json values while deserializing